src/ajax.js

```diff
@@ -557,8 +557,14 @@ jQuery.extend( {
 			}
 		}
 
+		// Uppercase the type
+		s.type = s.type.toUpperCase();
+
+		// Determine if request has content
+		s.hasContent = !rnoContent.test( s.type );
+
 		// Convert data if not already a string
-		if ( s.data && s.processData && typeof s.data !== "string" ) {
+		if ( s.data && ( s.processData || !s.hasContent ) && typeof s.data !== "string" ) {
```

Comment:
here is a sample for this problem: https://jsfiddle.net/1n5wo0bs/
in chrome debug "Network", you will find the request url is 
  https://fiddle.jshell.net/_display/dss.do?[object%20Object]